gamemode/sv_version.lua

```diff
@@ -14,41 +14,42 @@ function GM:CheckForNewVersion(ply)
 			print("Couldn't parse version file.")
 			return
 		end
-		local t = MsgClients()
-		if tab.version != GAMEMODE.Version then
-			t:Add("Out of date. ", Color(215, 20, 20))
-		end
-		t:Add("Latest version is " .. tab.version .. ". ", color_white)
+
+		local msg = {}
 		if tab.version != GAMEMODE.Version then
-			t:Add("Current version is " .. tostring(GAMEMODE.Version or "error") .. ".\n")
-			t:Add("Download the latest version from: " .. downloadlinks)
+			msg = {Color(215, 20, 20), "Out of date!\n",
+					Color(255, 222, 102), "You're on version ",
+					Color(0, 255, 0), (GAMEMODE.Version or "error"),
+					Color(255, 222, 102), " but the latest is ",
+					Color(0, 255, 0), tab.version, "\n",
+					Color(255, 222, 102), "Download the latest version from: ",
+					Color(11, 191, 227), downloadlinks}
 		else
-			t:Add("Up to date.")
+			msg = {Color(0, 255, 0), "Up to date!"}
 		end
-		t:Add("\n")
-		if ply then
-			t:Send(ply)
+
+		if IsValid(ply) then
+			ply:ChatMsg(unpack(msg))
 		else
-			t:Print()
-			local ct = ChatText(t.msgs)
-			for k, v in pairs(player.GetAll()) do
-				if v:IsListenServerHost() || v:IsSuperAdmin() then
-					ct:Send(v)
-				end
-			end
+			MsgC(unpack(msg))
+			MsgC("\n")
 		end
 	end
 	HTTP(req)
 end
 
 concommand.Add("ph_version", function (ply)
-	local t = MsgClients()
-	local n = GAMEMODE.Name or ""
-	t:Add(n .. " by Zikaeroh, " .. tostring(GAMEMODE.Version or "error") .. " (originally by MechanicalMind)\n", Color(255, 149, 129))
+	local msg = {Color(255, 149, 129),
+					(GAMEMODE.Name or "") ..
```

Comment:
Does this need to be a concat, or is just passing these as args fine?

Reply:
You're right, I could probably just separate these out and pass as arguments. I'll look into doing that.

Reply:
I have changed this to just be separate variables passed as arguments to the messaging methods.